src/thread_task.rs

```diff
@@ -791,14 +792,18 @@ unsafe fn delayed_fg_task<'l, 'ttd: 'l>(
 pub unsafe extern "C" fn rav1d_worker_task(data: *mut c_void) -> *mut c_void {
     let tc: &mut Rav1dTaskContext = &mut *(data as *mut Rav1dTaskContext);
     let c: &Rav1dContext = &*tc.c;
-    let ttd: &mut TaskThreadData = &mut *tc.task_thread.ttd;
+
+    // We clone the Arc here to avoid an immutable borrow of tc across the call
+    // to park
+    let ttd_clone = Arc::clone(&tc.task_thread.ttd);
```

Comment:
I might be forgetting something from a previous PR, but why can't we just access `ttd` through `tc` in `fn park`?  So we don't need a separate `&TaskThreadData` and extra the `Arc` clone and drop.

Reply:
I suppose we could. But I think we still need the extra clone, otherwise we have an immutable borrow (of ttd) across calls to `park`, even if we don't pass it as a parameter.

Reply:
Ah, okay, it's fine then